typing/rec_check.ml

```diff
@@ -984,6 +982,11 @@ and module_binding : (Ident.t * Typedtree.module_expr) -> bind_judg =
       let judg_E = modexp mexp << (Mode.join mM Guard) in
       Env.join (judg_E m) env
 
+and open_declaration : Typedtree.open_declaration -> bind_judg =
+  fun { open_expr = mexp; _ } m env ->
+      let judg_E = modexp mexp in
+      Env.join (judg_E m) env
```

Comment:
I think you also need to use `Env.remove_list` on the ids from `open_type` here. See the `Tstr_include` case above.